### PR TITLE
Amazon

### PR DESCRIPTION
**Pull request by 74956**

### CODE REVIEW

File: src/test/java/org/example/pages/CartPage.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public class CartPage extends InitDriver {
+
+    @FindBy(xpath = "//div[@id='sw-atc-details-single-container']//img")
+    private WebElement productImage;
+
+    @FindBy(xpath = "//div[@id='sw-atc-details-single-container']//span")
+    private WebElement msg;
+
+    @FindBy(xpath = "//div[@id='sw-atc-details-single-container']//div//i[@class='a-icon a-icon-alert']")
+    private WebElement icon;
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/CartPage.java:16]: Need to provide clear names

File: src/test/java/org/example/driver/Driver.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+public class Driver {
+
+    protected WebDriver driver;
+
+    @BeforeClass(alwaysRun = true)
+    public void openBrowser() {
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/driver/Driver.java:15]: Take a look at the Singleton pattern and change the driver initialization

File: src/test/java/org/example/pages/LoginPage.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package org.example.pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class LoginPage extends InitDriver {
+
+    @FindBy(xpath = "//div[@class='a-section']//input[@name='email']")
+    private WebElement fillEmail;
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/LoginPage.java:10]: do not use verbs for elements, 'emailfield' is better

File: src/test/java/org/example/pages/LoginPage.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    @FindBy(xpath = "//input[@type='submit']")
+    private WebElement submit;
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/LoginPage.java:13]: provide clear name

File: src/test/java/org/example/pages/LoginPage.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    @FindBy(xpath = "//div[@class='a-section']//input[@name='password']")
+    private WebElement fillPassword;
+
+    public LoginPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public void addEmail(String email) {
+        log.info("Add email");
+        fillEmail.sendKeys(email);
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/LoginPage.java:24]: need to add elt.clear() before elt.sendKeys() every time

File: src/test/java/org/example/pages/ProductPage.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    @FindBy(xpath = "//div//input[@name='submit.add-to-cart']")
+    private WebElement addToCartButton;
+
+    public ProductPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public int getQuantityOfItemsInTheCart() {
+        String str = itemsInTheCart.getText();
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/ProductPage.java:20]: do not use unclear names - 'str'

File: src/test/java/org/example/pages/ShoppingCartPage.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
+public class ShoppingCartPage extends InitDriver {
+
+    WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/ShoppingCartPage.java:14]: 20 is too much

File: src/test/java/org/example/pages/ShoppingCartPage.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+    @FindBy(xpath = "//div[@data-name='Active Cart']//div[@class='a-row']//*[contains(text(),'Shopping Cart')]")
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/ShoppingCartPage.java:16]: try to avoid finding by text because of localization

Comment (AliaksandraBarannik) [reply src/test/java/org/example/pages/ShoppingCartPage.java:16]: and simplify the locator

File: src/test/java/org/example/pages/ShoppingCartPage.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    private WebElement shoppingCart;
+
+    @FindBy(xpath = "//div[@class='a-fixed-left-grid']//img[contains(@src,'images') and @class='sc-product-image']")
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/ShoppingCartPage.java:19]: I didn't find anything using this locator. recheck it

File: src/test/java/org/example/pages/StartedPage.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class StartedPage extends InitDriver {
+
+    WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));
+
+    private final String URL = "https://www.amazon.com/ref=nav_logo";
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/StartedPage.java:15]: it should be better to put it in .properties file and read properties and use it

File: src/test/java/org/example/pages/StartedPage.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+    @FindBy(xpath = "//header//div//a[@data-nav-ref='nav_ya_signin']")
+    private WebElement clickOnSignInMenu;
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/pages/StartedPage.java:18]: remove click in the name

File: src/test/java/org/example/loggers/CartLogger.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
+package org.example.loggers;
+
+import org.example.pages.InitDriver;
+import org.example.pages.CartPage;
+import org.openqa.selenium.WebDriver;
+
+import java.util.logging.Logger;
+
+public class CartLogger extends InitDriver {
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/loggers/CartLogger.java:9]: Rename classes. CartLogger->CartPageService

File: src/test/java/org/example/loggers/LoginLogger.java [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+        log.info("Add password");
+        loginPage.addPassword(password);
+        return this;
+    }
+
+    public void logIn(User user) {
+        addEmail(user.getEmail());
+        clickOnSubmitButton();
+        addPassword(user.getPassword());
+        clickOnSubmitButton();
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/loggers/LoginLogger.java:41]: This user should return the service of new page. 
return new StartedPageService.
We don't need to intialize services in beforeClass/beforeTest. we initialize them in methods

File: src/test/java/org/example/driver/Driver.java [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+        options.addArguments("--headless");
+        options.addArguments("start-maximized");
+        driver = new ChromeDriver(options);
+        driver.get("https://www.amazon.com/");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void closeBrowser() {
+        driver.quit();
+        driver = null;
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/driver/Driver.java:27]: why do you use  driver = null here?

File: src/test/java/org/example/model/User.java [hunk 15]
```diff
@@ -0,0 +1,11 @@
+package org.example.model;
```
Comment (AliaksandraBarannik) [inline src/test/java/org/example/model/User.java:1]: rename the folder to object

File: src/test/java/org/example/pages/ShoppingCartPage.java (hunk 9 above)
Comment (74956) [reply src/test/java/org/example/pages/ShoppingCartPage.java:19]: using this locator i`m checking the list of items which were added to the shopping cart. Just add some item and click on the cart than you will see list of items

**Pull request merged**